onnxruntime/core/providers/cpu/ml/ml_common.h

```diff
@@ -289,57 +289,78 @@ static inline void ComputeSoftmaxZero(std::vector<float>& values) {
 }
 
 template <typename T>
-void write_scores(std::vector<T>& scores, POST_EVAL_TRANSFORM post_transform, int64_t write_index, Tensor* Z,
-                  int add_second_class) {
+static void write_scores(std::vector<T>& scores, POST_EVAL_TRANSFORM post_transform,
+                         T* Z, int add_second_class) {
   if (scores.size() >= 2) {
     switch (post_transform) {
       case POST_EVAL_TRANSFORM::PROBIT:
-        for (float& score : scores)
-          score = ComputeProbit(score);
+        for (auto it = scores.cbegin(); it != scores.cend(); ++it, ++Z)
+          *Z = static_cast<T>(ComputeProbit(static_cast<float>(*it)));
```

Comment:
nit: scores.cend() won't change so call it outside of the loop
for (auto it = scores.cbegin(), end = scores.cend(); it != end; ++it, ++z)